glib/src/subclass/types.rs

```diff
@@ -233,23 +234,33 @@ impl TypeData {
         self.parent_class
     }
 
-    /// Returns a pointer to the interface implementation specific data.
+    /// Returns a pointer to the class implementation specific data.
     ///
-    /// This is used for interface implementations to store additional data.
-    pub fn get_interface_data(&self, type_: ffi::GType) -> ffi::gpointer {
+    /// This is used for class implementations to store additional data.
+    pub fn get_class_data(&self, type_: ffi::GType) -> ffi::gpointer {
```

Comment:
```suggestion
    pub fn get_class_data(&self, type_: ffi::GType) -> Option<&dyn Any + Send + Sync> {
```

Reply:
Or even

```suggestion
    pub fn get_class_data<T: Any + Send + Sync + 'static>(&self, type_: ffi::GType) -> Option<&T> {
```

And have it also panic on type mismatch (because really must not happen!)

Reply:
> And have it also panic on type mismatch

Not sure how can I do something like this

Reply:
https://doc.rust-lang.org/nightly/std/any/trait.Any.html#method.is-2 for checking the type, https://doc.rust-lang.org/nightly/std/any/trait.Any.html#method.downcast_ref-2 and the mutable variant for getting the value out